File: DownloadLib/src/main/java/com/kronos/download/IDownloadDb.java
package com.kronos.download;

import java.util.HashMap;

/**
 * Created by Leif Zhang on 16/9/30.
 * Email leifzhanggithub@gmail.com
 */
public interface IDownloadDb {
    void saveToDb(HashMap<String, DownloadModel> models);

    HashMap<String, DownloadModel> getFromDB();
}

File: DownloadLib/src/main/java/com/kronos/download/IDownloadDb.kt
package com.kronos.download

import java.util.HashMap

/**
 * Created by Leif Zhang on 16/9/30.
 * Email leifzhanggithub@gmail.com
 */
interface IDownloadDb {

    fun saveToDb(model: DownloadModel?)

    fun saveToDb(models: HashMap<String, DownloadModel>?)

    fun getFromDB(config: DownloadConfig): HashMap<String, DownloadModel>
}
